pkcs11/pkcs11_test.go

```diff
@@ -594,3 +594,117 @@ func TestCreateCertificate(t *testing.T) {
 		t.Errorf("Returned certificate did not match loaded certificate")
 	}
 }
+
+func TestEncryptRSA(t *testing.T) {
+	msg := "Plain text to encrypt"
+	b := []byte(msg)
+	tests := []struct {
+		name string
+		bits int
+	}{
+		{"2048", 2048},
+		{"4096", 4096},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			s := newTestSlot(t)
+			o := keyOptions{RSABits: test.bits}
+			priv, err := s.generate(o)
+			if err != nil {
+				t.Fatalf("generate(%#v) failed: %v", o, err)
+			}
+			rsaPriv, ok := priv.(*rsaPrivateKey)
+			if !ok {
+				t.Fatalf("Private Key unexpected type, got %T, want *rsaPrivateKey", priv)
+			}
+			// SHA1 is the only hash function supported by softhsm
+			rsaPriv.WithHash(crypto.SHA1)
+			cipher, err := rsaPriv.encryptRSA(b)
+			if err != nil {
+				t.Errorf("encryptRSA Error: %v", err)
+				t.Log(cipher)
+			}
+		})
+	}
+}
+
+func TestDecryptRSA(t *testing.T) {
+	msg := "Plain text to encrypt"
+	b := []byte(msg)
+	tests := []struct {
+		name string
+		bits int
+	}{
+		{"2048", 2048},
+		{"4096", 4096},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			s := newTestSlot(t)
+			o := keyOptions{RSABits: test.bits}
+			priv, err := s.generate(o)
+			if err != nil {
+				t.Fatalf("generate(%#v) failed: %v", o, err)
+			}
+			rsaPriv, ok := priv.(*rsaPrivateKey)
+			if !ok {
+				t.Fatalf("Private Key unexpected type, got %T, want *rsaPrivateKey", priv)
+			}
+			rsaPriv.WithHash(crypto.SHA1)
+			cipher, err := rsaPriv.encryptRSA(b)
+			if err != nil {
+				t.Errorf("encryptRSA Error: %v", err)
+			}
+			decrypted, err := rsaPriv.decryptRSA(cipher)
+			if err != nil {
+				t.Errorf("decryptRSA Error: %v", err)
+			}
+			decrypted = bytes.Trim(decrypted,"\x00")
```

Comment:
PKCS#11 should be setting `cDecryptedLen` [here](https://github.com/lcforges/go-pkcs11/blob/my-work-so-far/pkcs11/pkcs11.go#L1833). 

Why is there NULL padding? Is it included or excluded in `cDecryptedLen`?